core/saltfilter/bloom.go

```diff
@@ -0,0 +1,34 @@
+// Package saltfilter used for reuse detection ONLY
+package saltfilter
+
+import (
+	"sync"
+
+	"github.com/willf/bloom"
+)
+
+var (
+	filterInstance *bloom.BloomFilter
+	itemCount      = 0
+	mux            = sync.Mutex{}
+)
+
+const itemSize = 1e6
+
+func init() {
+	filterInstance = bloom.NewWithEstimates(itemSize, 0.003)
+}
+
+func Add(data []byte) {
+	mux.Lock()
+	if itemCount > itemSize {
+		filterInstance.ClearAll()
```

Comment:
Come to think about it again, when you reset it the BF loses all its recent memories. You should try the double (or ping-pong) Bloom filters technique as used in ss-libev.

Reply:
What if use a bucket BF for more accuracy?

Reply:
What do you mean by "bucket BF"?

Reply:
actually, it is a Circle Bloom Filter: If current it full, then next one. Maybe regard as a complex PingPong Bloom Filter